src/amo/components/CollectionManager/index.js

```diff
@@ -183,6 +185,21 @@ export class CollectionManagerBase extends React.Component<Props, State> {
     }));
   };
 
+
+  handleSlug = (targetElement: Object) => {
+    const { name = '', value = '' } = targetElement;
+    const { creating } = this.props;
+    // only for add collection
+    if (!creating) return true;
+
+    if (name === 'name' && !this.state.userEnterSlug) {
+      this.setState({ slug: value.replace(/[^A-Za-z0-9]/g, '-') });
```

Comment:
It is a good start but we should maybe be a bit more defensive, see when I enter empty spaces:

![2018-05-07 11 51 47](https://user-images.githubusercontent.com/217628/39696213-4809fa8a-51ed-11e8-93e9-85b4bc97223b.gif)